tests/components/auth/test_init.py

```diff
@@ -606,18 +624,34 @@ async def test_ws_delete_all_refresh_tokens(
 
     tokens = result["result"]
 
-    await ws_client.send_json(
-        {
-            "id": 6,
-            "type": "auth/delete_all_refresh_tokens",
-        }
-    )
+    ws_json = {
+        "id": 6,
+        "type": "auth/delete_all_refresh_tokens",
+    }
+    if delete_token_type:
+        ws_json["token_type"] = delete_token_type
+
+    await ws_client.send_json(ws_json)
 
     result = await ws_client.receive_json()
     assert result, result["success"]
+    remaining_tokens = 0
     for token in tokens:
         refresh_token = await hass.auth.async_get_refresh_token(token["id"])
-        assert refresh_token is None
+        if refresh_token is not None:
+            remaining_tokens += 1
+
+        if delete_token_type is None or token["type"] == delete_token_type:
```

Comment:
Sorry for the lack of clarity. The goal was not to have an if in the body of the test

Reply:
Hmm, that's a pretty unfamiliar concept to me.

I don't think it's likely I'm going to stumble on the approved syntax here... any chance you could elaborate a bit what I should have done instead? Nothing obvious is jumping out to me as a better way I could have written this highlight.